atomic_reactor/plugins/exit_store_metadata_in_osv3.py

```diff
@@ -222,7 +222,9 @@ def run(self):
         pulp_pull_results = self.workflow.postbuild_results.get(PulpPullPlugin.key)
         if pulp_pull_results:
             _, media_types = pulp_pull_results
-            annotations['media-types'] = json.dumps(media_types)
+            if pulp_push_results:
+                media_types = list(set(media_types + ['application/json']))
```

Comment:
I don't really like setting the v1 content media type here, but I can't really think of an alternative :(

Reply:
Wait, is this code path only for arrangement 3? It requires both `pulp_pull` and `pulp_push` to run on the same build. In arrangement 4, this is no longer true. If so, shouldn't pulp_pull already include `'application/json'` media type?

Reply:
This one is for both:
* in arrangement 3 the build annotations would be read by `koji_upload` from this build
* in arrangement 4 worker build's `pulp_push` would write the annotation to the worker build. 
In `koji_import` this annotation from x86-64 would be combined with `pulp_pull` result

Reply:
Yes, this looks like a fine approach.

Reply:
> in arrangement 4 worker build's pulp_push would write the annotation to the worker build.

I must be missing something, but the conditional for pulp_push_results is nested under pulp_pull_results. This is a no-op in arrangement 4 since pulp_pull and pulp_push do not run on the same build.

Reply:
Agree, implemented independent setting of media types. The tests cannot properly verify that (as those have a single instance of plugins list) though